### PR TITLE
feat: refacto proto generation, support optional field

### LINKED ISSUE #127: generateproto not generating int fields
Hello and hope you're doing well!

I started using this and ran into a small issue. I created a model where I'm using an integer field for state with a list of choices.
Screenshot 2023-01-23 at 5 45 19 PM

However, any time I run `python manage.py generateproto` it creates the state field as a string in the proto file. 
Screenshot 2023-01-23 at 5 48 15 PM

Is this intended behavior or a known issue? I've been working around it by generating proto files with `python -m grpc_tools.protoc` and manually updating the import statements, but it would be nice if the `generateproto` command could do this. If I'm missing anything, let me know and thanks in advance for the help!

### PR DESCRIPTION
**Pull request by legau**
Closes #127

### CODE REVIEW

File: django_socio_grpc/protobuf/proto_classes.py [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+
+StructMessage = ProtoMessage(
+    base_name="google.protobuf.Struct",
+    fields=[],
+    imported_from="google/protobuf/struct.proto",
+)
+
+
+@dataclass
+class ProtoRpc:
```
Comment (AMontagu) [inline django_socio_grpc/protobuf/proto_classes.py:588]: add comment to explain the purpose of this class

File: django_socio_grpc/protobuf/proto_classes.py [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+        messages = {}
+        if isinstance(self.request, ProtoMessage):
+            messages.update(self.request.get_all_messages())
+        if isinstance(self.response, ProtoMessage):
+            messages.update(self.response.get_all_messages())
+        return messages
+
+
+@dataclass
+class ProtoService:
```
Comment (AMontagu) [inline django_socio_grpc/protobuf/proto_classes.py:613]: add comment to explain the purpose of this class

File: django_socio_grpc/protobuf/proto_classes.py [hunk 3]
```diff
@@ -0,0 +1,713 @@
...
+    @property
+    def field_line(self) -> str:
+        values = [self.field_type_str, self.name, f"= {self.index};"]
+        if self.cardinality != FieldCardinality.NONE:
+            values.insert(0, self.cardinality.value)
+        return " ".join(values)
+
+    @property
+    def has_serializer_field(self) -> bool:
+        return isinstance(self.field_type, type) and issubclass(
```
[the remark this answers is not in the document]
Comment (legau) [reply django_socio_grpc/protobuf/proto_classes.py:91]: type is type, this is to check if field_type is a class to be able to call `issubclass`

File: django_socio_grpc/protobuf/proto_classes.py [hunk 4]
```diff
@@ -0,0 +1,713 @@
...
+
+    @classmethod
+    def _get_cardinality(self, field: serializers.Field):
+        return FieldCardinality.OPTIONAL if field.allow_null else FieldCardinality.NONE
+
+    @classmethod
+    def from_field_dict(cls, field_dict: FieldDict) -> "ProtoField":
+        cardinality = FieldCardinality.NONE
+        field_type = field_dict["type"]
+        type_parts = field_type.split(" ")
```
[the remark this answers is not in the document]
Comment (legau) [reply django_socio_grpc/protobuf/proto_classes.py:103]: would be better indeed

File: django_socio_grpc/protobuf/proto_classes.py [hunk 5]
```diff
@@ -0,0 +1,713 @@
...
+    base_name: str
+    fields: List["ProtoField"] = dataclass_field(default_factory=list)
+    comments: Optional[List[str]] = None
+    serializer: Optional[serializers.BaseSerializer] = None
+    imported_from: Optional[str] = None
+    suffixable: bool = True
+    prefixable: bool = True
+
+    prefix: str = ""
+    suffix: ClassVar = ""
```
[the remark this answers is not in the document]
Comment (legau) [reply django_socio_grpc/protobuf/proto_classes.py:317]: Suffix is set as a class variable (see RequestMessage with `Request` suffix)

ClassVar is needed by dataclasses to differentiate between the 2

**Pull request merged**